### PR TITLE
fix: separate model and component definition

### PR DESCRIPTION
**Pull request by AlexLengyel**
Splitting Component Definition & Model Definition

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sharanyavinod** [comment]
@AlexLengyel Shouldn't we remove the model from the component-definition once the corresponding changes are complete in the universal editor as well? As it is now, editor will break until the new PR there is merged.

**Comment by AlexLengyel** [comment]
> @AlexLengyel Shouldn't we remove the model from the component-definition once the corresponding changes are complete in the universal editor as well? As it is now, editor will break until the new PR there is merged.

I am not planning to merge it until the required changes in UE are not done.

**Comment by sharanyavinod** [comment]
@AlexLengyel Wouldn't that make it harder to test the new model definition?

**Comment by sharanyavinod** [comment]
Why not introduce the new model definition, work with it in the cors and editor updates and once everything is synced and model in component definition is not used anymore, easily remove it? Wdyt?

File: build/static/model-definition.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  {
+    "uri": "",
+    "fields": []
+  },
+  {
+    "modeltype": "container",
+    "fields": []
+  },
+  {
```
Comment (sharanyavinod) [inline build/static/model-definition.json:42]: Every model object should have an id. If there is no id, please remove the corresponding object from the array here.

**Review by sharanyavinod — APPROVED** [review]

**Pull request merged**